### ISSUE TITLE
Set up Cloudflare purging when publishing/editing blog posts

### ISSUE THREAD

**Issue by geerlingguy** [geerlingguy/jeffgeerling-com#146]
Any time I publish or save an edit to a blog post, I'd like to automatically trigger a Cloudflare purge on the following paths:

  1. The blog post URL
  2. The home page (/)
  3. The blog page (/blog)

Might not always need to do the latter two, but it's safe enough to do anyways, IMO.

One thing to consider—the cache might still be stale in Nginx itself, since it's storing a 5-10 minute separate cache of everything. So one thing I could do is disable that cache since I'm routing _everything_ through Cloudflare now. Maybe.

Related to #141.

**Comment by minecraftchest1** [comment]
I have been running a WordPress blog behind Cloudflare for a few days and made a few changes with Cloudflare set to provide a 4 hour cache time in both the cache page in the Cloudflare Dashboard, as well as with a page rule and it has always updated immediately for me. From that, I don't think that caching will be a problem for you. Granted, I haven't played with it very much and I am using WordPress, not Drupal so your results may very.

TLDR: I would check to see if caching is actually going to be an issue before trying to do automatic cache clearing.

**Comment by geerlingguy** [comment]
I have a 1 hour TTL set in Cloudflare, and it definitely respects that—I'd rather make sure the RSS feed and home page are updated immediately, so the Drupal module integrates with Cloudflare's API to purge by path, and that should hopefully be enough for me.

The one thing I need to look into is if I can get that API functionality on a free plan. Not sure if I'd need Pro or Enterprise to do that :/

I remember it being easier to set up with Varnish or Fastly (which uses Varnish IIRC).

**Comment by minecraftchest1** [comment]
I haven't had issues with using the api on the cloudflare free plan, but I haven't played with it much.

**Comment by userbradley** [comment]
The cloudflare free plan will be fine.

[As long as you don't send more than 1200 requests per 5 minutes](https://api.cloudflare.com/#getting-started-requests:~:text=cfa33fe60e8e34073c149323454383fc9005d25c9b4c502c2f063457ef65322eade065975001a0b4b4c591c5e1bd36a6e8f7e2d4fa8a9ec01c64c041e99530c2%2D07b9efe0acd78c82c8d9c690aacb8656d81c369246d7f996a205fe3c18e9254a%22%60-,Rate%20limiting,-The%20Cloudflare%20API)

I am looking at doing something similar for my [site](https://documentation.breadnet.co.uk) 

I'll take a stab at this, and let you know how simple/ annoying it was

**Comment by geerlingguy** [comment]
For now I've actually set this up using an Ansible playbook, since the purge also has to happen _after_ Nginx static cache is cleared, and I'd rather not do all that through Drupal/PHP.
